amulet/deployer.py

```diff
@@ -429,7 +428,8 @@ def setup(self, timeout=600, cleanup=True):
             with self.deploy_w_timeout(timeout):
                 subprocess.check_call(shlex.split(cmd))
 
-        self.sentry = Talisman(self.services, timeout=timeout)
+        self.sentry = Talisman(self.services)
+        self.sentry.wait_for_status(self.sentry.juju_env, self.services, timeout)
```

Comment:
I was trying to preserve the implicit wait in Deployment.setup() because it makes sense to me (there's already an implicit wait built in to juju-deployer), and to try to keep this <2.0 possible.  Also, I expect most tests will want to move to either wait_for_statuses or wait_for_messages.

If we're not concerned about keeping this pre-2.0, then gutting all the existing wait methods and making a single wait() that performs all of the existing checks of wait_for_status and wait, as well as optionally performing either or both of wait_for_statuses and wait_for_messages makes sense, and dropping all implicit waiting entirely (except for juju-deployer, of course).

In that case, we could also consider keeping the initial timeout and using that as the default all the way through instead of always falling back to the 300s default.